### PR TITLE
Feat/dsm announcements [JIRA: MATD-52]

### PR DESCRIPTION
**Pull request by Abhishek-Kharpal**
* API Endpoints for get, post, update and delete is created for dsm-announcements.
* Swagger documentation for all the endpoints are created.
* Validation was done using middleware and Joi.
Screenshot 2023-02-17 at 1 46 00 AM
For GET: 

Screenshot 2023-02-17 at 1 46 36 AM

For POST:
Screenshot 2023-02-17 at 1 47 59 AM

For GET by ID:
Screenshot 2023-02-17 at 1 48 25 AM

For PATCH:
Screenshot 2023-02-17 at 1 49 44 AM

For DELETE:

Screenshot 2023-02-17 at 1 50 10 AM

### CODE REVIEW

File: src/routes/api/dsm/announcements.routes.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+*/
+
+router.route('/')
+  .get(listAnnouncements)
+  .post(generateValidationMiddleware(announcementsSchema.createAnnouncementSchema), createAnnouncement);
+
+const parseIntIdParam = (req, res, next) => {
+  req.params.id = parseInt(req.params.id, 10);
+  next();
+};
```
Comment (ritikrajdev) [inline src/routes/api/dsm/announcements.routes.js:94]: can we shift it in middleware generator that can parse and set any req param to required datatype ?

File: src/services/dsm/announcements.services.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  }
+  return announcement;
+};
+
+/** 
+  * Service to delete announcement by id
+  * @param {Number} announcementId - Announcement id
+  * @returns {void} - No return value
+  * @throws {HttpError} - Throws an error if announcement not found
+*/
```
Comment (ritikrajdev) [inline src/services/dsm/announcements.services.js:91]: Really a good way to describe function.
Great Work !!

**Review by ritikrajdev — APPROVED** [review]

File: src/services/dsm/announcements.services.js (hunk 2 above)
Comment (Abhishek-Kharpal) [reply src/services/dsm/announcements.services.js:91]: Thanks :)

File: src/routes/api/dsm/announcements.routes.js (hunk 1 above)
Comment (Abhishek-Kharpal) [reply src/routes/api/dsm/announcements.routes.js:94]: Sure, I can try that.

Comment (Abhishek-Kharpal) [reply src/routes/api/dsm/announcements.routes.js]: Done in commits: https://github.com/tech-university-india/abd-backend/pull/21/commits/291ce72eac080f75a47450a6efeda213f2370489 and  https://github.com/tech-university-india/abd-backend/pull/21/commits/9bad5bbf34f30bda71ae7c25eb13aba51991d578

Comment (ritikrajdev) [reply src/routes/api/dsm/announcements.routes.js]: Great work !!

File: src/schemas/dsm/announcementsSchema.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    .number()
+    .integer()
+    .min(1)
+});
+
+const patchAnnouncementSchema = joi.object({
+  content: joi
+    .string()
+    .min(1)
+    .max(255)
```
Comment (sugumar-mani) [inline src/schemas/dsm/announcementsSchema.js:22]: we may need to think of this char limit in future.

**Review by sugumar-mani — APPROVED** [review]

File: src/schemas/dsm/announcementsSchema.js (hunk 3 above)
Comment (Abhishek-Kharpal) [reply src/schemas/dsm/announcementsSchema.js]: Sure, I have raised the value from 255 to 1500 for now.

**Pull request merged**